// Protocol messages for compiler IRs.
//
// Copyright 2019-2020 the ProGraML authors.
//
// Contact Chris Cummins <chrisc.101@gmail.com>.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package programl;

option cc_enable_arenas = true;
option go_package = "programlpb";
option java_multiple_files = true;
option java_outer_classname = "IrProto";
option java_package = "com.programl";

// A compiler intermediate representation.
message Ir {
  enum Type {
    UNKNOWN = 0;
    LLVM = 1;
    XLA_HLO = 2;
  }
  Type type = 1;
  // The compiler version, as a single integer. Major + minor versions
  // should be converted to this single number, e.g. 6.0.0 -> 600.
  int64 compiler_version = 2;
  // The command that was used to produce this IR.
  string cmd = 3;
  // The text of the IR.
  string text = 4;
}

// A list of compiler IRs.
message IrList {
  repeated Ir ir = 1;
}
